packages/react-query/src/useBaseQuery.ts

```diff
@@ -131,6 +136,26 @@ export function useBaseQuery<
     result,
   )
 
+  if (
+    defaultedOptions.experimental_promise &&
+    !isServer &&
+    willFetch(result, isRestoring) &&
+    !observer.hasListeners()
+  ) {
+    const promise = isNewCacheEntry
+      ? // Fetch immediately on mount in order to ensure `.promise` is resolved even if the component is unmounted
```

Comment:
NIT: This comment has me a bit confused, is it "in render" instead of "on mount"?

Reply:
```suggestion
      ? // Fetch immediately on render in order to ensure `.promise` is resolved even if the component is unmounted
```